service/command.go

```diff
@@ -16,12 +16,59 @@ package service // import "go.opentelemetry.io/collector/service"
 
 import (
 	"errors"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
 
 	"github.com/spf13/cobra"
 
+	"gopkg.in/yaml.v3"
+
+	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/featuregate"
 )
 
+type componentsOutput struct {
+	BuildInfo  component.BuildInfo
+	Receivers  []config.Type
+	Processors []config.Type
+	Exporters  []config.Type
+	Extensions []config.Type
+}
+
+// newBuildCommand constructs a new cobra.Command sub command using the given CollectorSettings.
+func newBuildSubCommand(set CollectorSettings) *cobra.Command {
```

Comment:
please extract this in its own file with tests as well.

Reply:
Thanks please I have written test for this already: https://github.com/open-telemetry/opentelemetry-collector/pull/6322/files#diff-ef234bff50b971d3085412fc8ca957dde031204f5c4591a8833611f9758cb36e. Are you saying that I should create a new file for this s well as for the test respectively?

Reply:
Indeed :) "command_build_info.go" and "command_build_info_test.go"